coordinator/microbench/README.md

```diff
@@ -0,0 +1,68 @@
+# Stargate Micro-benchmarking
+
+This module contains [JMH](https://github.com/openjdk/jmh) benchmarks for Stargate.
+It uses the [jmh-maven-plugin](https://github.com/metlos/jmh-maven-plugin) to run the benchmarks, primarily from the command line.
+
+## Running benchmarks
+
+> **NOTE:** It's advised that before running the benchmarks you build the project with `./mvnw clean install -DskipTests` to ensure that the latest changes are included. 
+
+To run all benchmarks, ensure you are in the `microbench` directory and run:
+
+```bash
+../mvnw jmh:benchmark
+```
+
+To run a specific benchmark class:
+
+```bash
+../mvnw jmh:benchmark -Djmh.benchmarks=MyBenchmark
+```
+
+The JHM framework allows specification of different parameters for a benchmark run.
+Check how the parameter passing works in the [jmh-maven-plugin](https://github.com/metlos/jmh-maven-plugin#passing-parameters) documentation.
+
+## Using profilers
```

Comment:
Maybe this section should include warning for Windows and macOS users.. Smth like:

`Don't try this on Windows machine, and expect limited options on your macOS.`

:smile: